src/document-click.js

```diff
@@ -248,12 +243,29 @@ export function onDocumentElementClick_(
   // Look for the referenced element.
   const hash = tgtLoc.hash.slice(1);
   let elem = null;
+
+  // Removal of the targeted attribute here to disable browsers' default
+  // scrolling behavior and rely on viewport./*OK*/scrollIntoView(elem)
+  // call to scroll to it.
+  // Without doing this, the viewport will not catch the updated scroll
+  // position on iOS Safari and hence calculate the wrong scrollTop for
+  // the scrollbar jumping the user back to the top for failing to calculate
+  // the new jumped offset.
+  // See https://github.com/ampproject/amphtml/issues/5334 for more details.
+  let oldAttribute = {};
   if (hash) {
     elem = ampdoc.getRootNode().getElementById(hash);
     if (!elem) {
       // Fallback to anchor[name] if element with id is not found.
       // Linking to an anchor element with name is obsolete in html5.
       elem = ampdoc.getRootNode().querySelector(`a[name=${hash}]`);
+      if (elem) {
+        oldAttribute = {key: 'name', value: elem.name};
+        elem.removeAttribute('name');
+      }
+    } else {
+      oldAttribute = {key: 'id', value: elem.id};
+      elem.removeAttribute('id');
```

Comment:
We need to ensure there's not a `[name=${hash}]` element, too. And there could be multiple of both.